### PR TITLE
Refactor Python 3

### PR DESCRIPTION
**Pull request by pFernbach**
Fix previous refactoring (see discussions in https://github.com/loco-3d/multicontact-locomotion-planning/pull/28)

### CODE REVIEW

**Review by nim65s — APPROVED** [review]

File: scripts/mlp/centroidal/topt.py [hunk 1]
```diff
@@ -58,262 +60,275 @@ def isNewPhaseFromCS(cs_com,cs_initGuess,p_id):
...
     s_init = s_final.copy()
     s_init[2] -= cfg.COM_SHIFT_Z
     phase_init.init_state = s_init
     phase_init.final_state = s_final
-    genSplinesForPhase(phase_init,0.,cfg.TIME_SHIFT_COM)
+    genSplinesForPhase(phase_init, 0., cfg.TIME_SHIFT_COM)
     # fill the rest of the phase with the trajectory in previous_phase
-    for i in range(1,len(prev_phase_init.time_trajectory)): # values for id = 0 are aready the last point in the trajectory computed by genSplinesForPhase
+    for i in range(1, len(prev_phase_init.time_trajectory)
+                   ):  # values for id = 0 are aready the last point in the trajectory computed by genSplinesForPhase
```
Comment (nim65s) [inline scripts/mlp/centroidal/topt.py]: This comment should go on its own line

File: scripts/mlp/centroidal/topt.py [hunk 2]
```diff
@@ -58,262 +60,275 @@ def isNewPhaseFromCS(cs_com,cs_initGuess,p_id):
...
-        patch = getPhasePatchforEE(p0,ee)
-        tp.setInitialPose(isContactEverActive(cs,ee), patch.placement.translation, patch.placement.rotation, ee)
-    tp.setMass(cfg.MASS);#FIXME
-    tp.getMass();
-    
-    #add all effector phases to the problem : 
+        patch = getPhasePatchforEE(p0, ee)
+        tp.setInitialPose(isContactEverActive(cs, ee), patch.placement.translation, patch.placement.rotation, ee)
+    tp.setMass(cfg.MASS)
+    #FIXME
```
Comment (nim65s) [inline scripts/mlp/centroidal/topt.py:309]: I think there is a missing space after the `#`, but yapf should have added it.

File: scripts/mlp/contact_sequence/manually_defined/flat_floor.py [hunk 3]
```diff
@@ -22,23 +21,24 @@
...
+placement = cs.contact_phases[0].RF_patch.placement.copy()  # take the previous position of the right feet
 pos = placement.translation
-pos[0] += 0.15 # move of 15cm in the x direction
+pos[0] += 0.15  # move of 15cm in the x direction
 placement.translation = pos
-moveEffectorToPlacement(fb,v,cs,fb.rfoot,placement) # add a new contact phase, with a contact created for the right feet at the given placement
-
+moveEffectorToPlacement(
+    fb, v, cs, fb.rfoot,
+    placement)  # add a new contact phase, with a contact created for the right feet at the given placement
```
Comment (nim65s) [inline scripts/mlp/contact_sequence/manually_defined/flat_floor.py]: This comment should go on its own line

File: scripts/mlp/contact_sequence/sl1m.py [hunk 4]
```diff
@@ -42,7 +42,8 @@ def right_foot_constraints(transform):
 def left_foot_constraints(transform):
     global __ineq_left_foot
     if __ineq_left_foot is None:
-        filekin =os.environ["INSTALL_HPP_DIR"]+"/share/talos-rbprm/com_inequalities/feet_quasi_flat/talos_COM_constraints_in_LF_effector_frame_REDUCED.obj"
+        filekin = os.environ[
+            "INSTALL_HPP_DIR"] + "/share/talos-rbprm/com_inequalities/feet_quasi_flat/talos_COM_constraints_in_LF_effector_frame_REDUCED.obj"
```
Comment (nim65s) [inline scripts/mlp/contact_sequence/sl1m.py]: still too long, as well as the 2 others below.

File: scripts/mlp/demo_configs/common_talos.py [hunk 5]
```diff
@@ -42,24 +42,54 @@
...
+    ]  #head
+).T
 
 masks_posture = np.matrix(np.ones(32)).T
 #masks_posture[:11] = 0
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/common_talos.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_bauzil_with_stairs.py [hunk 6]
```diff
@@ -3,18 +3,18 @@
...
-
+p_max = 0.13  #setting used to compute the default height of the effector trajectory. end_effector/bezier_predef.py : computePosOffset()
 
 GUIDE_STEP_SIZE = 0.6
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
-
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_bauzil_with_stairs.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_lp_complex.py [hunk 7]
```diff
@@ -4,25 +4,24 @@
...
 #USE_WP_COST = False # use wp from the contact sequence in the cost function of the centroidal solver
 
-
 EFF_T_PREDEF = 0.2
 p_max = 0.13
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_lp_complex.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_lp_maze.py [hunk 8]
```diff
@@ -17,17 +15,16 @@
...
 
-
 #EFF_T_PREDEF = 0.2
 #p_max = 0.08
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
-
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_lp_maze.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_lp_slalom.py [hunk 9]
```diff
@@ -3,25 +3,24 @@
...
 #USE_WP_COST = False # use wp from the contact sequence in the cost function of the centroidal solver
 
-
 EFF_T_PREDEF = 0.3
 p_max = 0.2
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_lp_slalom.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_mazeEas.py [hunk 10]
```diff
@@ -16,5 +16,6 @@
 
 GUIDE_STEP_SIZE = 0.7
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
-
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_mazeEas.py]: This comment should go on its own line

File: scripts/mlp/demo_configs/talos_mazeEas_oriented.py [hunk 11]
```diff
@@ -16,5 +16,6 @@
 
 GUIDE_STEP_SIZE = 0.7
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
-
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_mazeEas_oriented.py]: … you get the idea x)

File: scripts/mlp/demo_configs/talos_navBauzil.py [hunk 12]
```diff
@@ -19,5 +18,6 @@
 
 GUIDE_STEP_SIZE = 0.6
 
-IK_REFERENCE_CONFIG = np.matrix(Robot.referenceConfig_elbowsUp).T #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
-
+IK_REFERENCE_CONFIG = np.matrix(
+    Robot.referenceConfig_elbowsUp
+).T  #Reference config used by the wholeBody script, may be different than the one used by the planning (default value is the same as planning)
```
Comment (nim65s) [inline scripts/mlp/demo_configs/talos_navBauzil.py]: same here

File: scripts/mlp/end_effector/bezier_constrained.py [hunk 13]
```diff
@@ -602,165 +693,180 @@ def computeProblemConstraints(pData,fullBody,pathId,t,eeName,viewer):
...
     # approximate the switching times (infos from limb-rrt)
-    if len(t_paths)>2 :
-        splits=[]
-        t_ratio = t/t_paths[-1] # ratio between the imposed time of the bezier curve (t) and the "time" (a pseudo distance) of the solution of the rrt 
-        for i in range(1,len(t_paths)-1):
-            ti = t_paths[i]*t_ratio
+    if len(t_paths) > 2:
+        splits = []
+        t_ratio = t / t_paths[
+            -1]  # ratio between the imposed time of the bezier curve (t) and the "time" (a pseudo distance) of the solution of the rrt
```
Comment (nim65s) [inline scripts/mlp/end_effector/bezier_constrained.py]: here too

File: scripts/mlp/utils/util.py [hunk 14]
```diff
@@ -56,31 +59,34 @@ def MotiontoVec(M):
...
-        rot[:,1]=np.asarray(vect[6:9]).reshape(-1)
-        rot[:,2]=np.asarray(vect[9:12]).reshape(-1)
-    else :
-        rot[:,0]=vect[3:6]
-        rot[:,1]=vect[6:9]
-        rot[:,2]=vect[9:12]
-    placement.rotation =rot
+    if len(
+            rot[:, 0].shape
+    ) == 1:  # depend if eigenpy.switchToNumpyArray() have been called, FIXME : there should be a better way to check this
```
Comment (nim65s) [inline scripts/mlp/utils/util.py]: and here

File: scripts/mlp/utils/wholebody_result.py [hunk 15]
```diff
@@ -1,246 +1,303 @@
...
-        self.zmp_t = np.matrix(np.zeros([3,N])) # Zero Moment Point location at each time step(3 x N )
-        self.zmp_reference = np.matrix(np.zeros([3,N])) # Zero Moment Point location at each time step(3 x N ) as computed by the CoM trajectory optimization
-        self.waist_orientation_reference = np.matrix(np.zeros([4,N])) # Reference used (from contact planning) of the waist orientation, stored as quaternion (x,y,z,w)        
-        self.d_waist_orientation_reference = np.matrix(np.zeros([3,N])) # Reference used (from contact planning) of the waist angular velocity, stored as angular velocity (in world frame)                
-        self.dd_waist_orientation_reference = np.matrix(np.zeros([3,N])) # Reference used (from contact planning) of the waist angular acceleration, stored as angular acceleration (in world frame)              
-        self.eeNames = eeNames # The list of all effector names used during the motion
+        self.t_t = np.array([t_begin + i * self.dt for i in range(N)
+                             ])  # timing vector (1 x N ) (time at each discretized point, from t=0 to t=dt*(N-1))
```
Comment (nim65s) [inline scripts/mlp/utils/wholebody_result.py]: same in this file

File: scripts/mlp/wholebody/tsid_invdyn.py [hunk 16]
```diff
@@ -446,152 +466,160 @@ def stopHere():
...
 
     if cfg.PLOT:
         from mlp.utils import plot
-        plot.plotEffectorRef(stored_effectors_ref,dt)   # plot inside this file, as we have access to the real Bezier object and not only the discretization stored in 'res' struct         
-    
-    assert (k_t == res.N-1) and "res struct not fully filled."
-    return res,pinRobot
+        plot.plotEffectorRef(
+            stored_effectors_ref, dt
+        )  # plot inside this file, as we have access to the real Bezier object and not only the discretization stored in 'res' struct
```
Comment (nim65s) [inline scripts/mlp/wholebody/tsid_invdyn.py]: also here

**Review by nim65s — CHANGES_REQUESTED** [review]

**Comment by pFernbach** [comment]
Thanks for the review, latest commit apply the changes requested.

**Review by nim65s — APPROVED** [review]
That's looking better, thanks :)

**Pull request merged**